etna/datasets/hierarchical_structure.py

```diff
@@ -33,7 +34,9 @@ def __init__(self, level_structure: Dict[str, List[str]], level_names: Optional[
         tree_depth = len(hierarchy_levels)
 
         self.level_names = self._get_level_names(level_names, tree_depth)
-        self._level_series: Dict[str, List[str]] = {self.level_names[i]: hierarchy_levels[i] for i in range(tree_depth)}
+        self._level_series: Dict[str, Tuple[str, ...]] = {
```

Comment:
Let's use the same typing for segments as in TSDataset [here](https://github.com/tinkoff-ai/etna/blob/c04bf64b71b740abf880a37fe70c36b68ba113d1/etna/datasets/tsdataset.py#L439)

Reply:
Previous version wasn't safe enough, because we returned reference to mutable sequence. In fact, we have 2 options: return reference to immutable sequance (current version) or return a copy (like in the linked example).

Reply:
Ok, let's return copy here to keep the consistency

Reply:
In this case, we sacrifice performance for consistency.